src/scores/probability/brier_impl.py

```diff
@@ -61,3 +67,155 @@ def brier_score(
         check_binary(obs, "obs")
 
     return mse(fcst, obs, reduce_dims=reduce_dims, preserve_dims=preserve_dims, weights=weights)
+
+
+def ensemble_brier_score(
+    fcst: XarrayLike,  # type: ignore
+    obs: XarrayLike,  # type: ignore
+    ensemble_member_dim: str,
+    thresholds: Union[float, int, Sequence[float]],
+    *,  # Force keywords arguments to be keyword-only
+    reduce_dims: Optional[FlexibleDimensionTypes] = None,
+    preserve_dims: Optional[FlexibleDimensionTypes] = None,
+    weights: Optional[xr.DataArray] = None,
+    fair_correction: bool = True,
+    threshold_mode: Literal[">=", ">"] = ">=",
+) -> XarrayLike:  # type: ignore
+    """
+    Calculates the Brier score for an ensemble forecast for the provided thresholds. By default,
+    the fair Brier score is calculated, which is a modified version of the Brier score that
+    applies a correction related to the number of ensemble members and the number of
+    ensemble members that predict the event.
+
+    The fair Brier score for ensembles is defined for a single event threshold as:
+
+
+    .. math::
+        s_{i,y} = \\left(\\frac{i}{m} - y\\right)^2 - \\frac{i(m-i)}{m^2(m-1)}
+
+    Where:
+
+    - :math:`i` is the number of ensemble members that predict the event
+    - :math:`m` is the total number of ensemble members
+    - :math:`y` is the observed event
```

Comment:
I think I'd prefer `y` is the binary observation (i.e., y=1 if the event occurred and 0 otherwise)

Usually in this content "event" refers to one particular outcome out of two possible outcomes (e.g. exceeding some given threshold)